### ISSUE TITLE
src/sample.c: couple of coverity findings

### ISSUE THREAD

**Issue by chipitsine** [haproxy/haproxy#2484]
### Tool Name and Version

coverity

### Code Report

```plain
** CID 1539957:    (CHECKED_RETURN)
/src/sample.c: 4266 in sample_conv_jwt_verify_check()
/src/sample.c: 4265 in sample_conv_jwt_verify_check()


________________________________________________________________________________________________________
*** CID 1539957:    (CHECKED_RETURN)
/src/sample.c: 4266 in sample_conv_jwt_verify_check()
4260     
4261     #ifdef USE_OPENSSL
4262     static int sample_conv_jwt_verify_check(struct arg *args, struct sample_conv *conv,
4263                                            const char *file, int line, char **err)
4264     {
4265            vars_check_arg(&args[0], NULL);
>>>     CID 1539957:    (CHECKED_RETURN)
>>>     Calling "vars_check_arg" without checking return value (as is done elsewhere 12 out of 15 times).
4266            vars_check_arg(&args[1], NULL);
4267     
4268            if (args[0].type == ARGT_STR) {
4269                    enum jwt_alg alg = jwt_parse_alg(args[0].data.str.area, args[0].data.str.data);
4270     
4271                    if (alg == JWT_ALG_DEFAULT) {
/src/sample.c: 4265 in sample_conv_jwt_verify_check()
4259     }
4260     
4261     #ifdef USE_OPENSSL
4262     static int sample_conv_jwt_verify_check(struct arg *args, struct sample_conv *conv,
4263                                            const char *file, int line, char **err)
4264     {
>>>     CID 1539957:    (CHECKED_RETURN)
>>>     Calling "vars_check_arg" without checking return value (as is done elsewhere 12 out of 15 times).
4265            vars_check_arg(&args[0], NULL);
4266            vars_check_arg(&args[1], NULL);
4267     
4268            if (args[0].type == ARGT_STR) {
4269                    enum jwt_alg alg = jwt_parse_alg(args[0].data.str.area, args[0].data.str.data);
4270     

** CID 1539956:  Uninitialized variables  (UNINIT)
/src/ssl_sample.c: 396 in sample_conv_aes_gcm()


________________________________________________________________________________________________________
*** CID 1539956:  Uninitialized variables  (UNINIT)
/src/ssl_sample.c: 396 in sample_conv_aes_gcm()
390                     ret = a2base64(trash->area, 16, aead_tag.data.u.str.area, aead_tag.data.u.str.size);
391                     if (ret < 0)
392                             goto err;
393     
394                     aead_tag.data.u.str.data = ret;
395                     aead_tag.data.type = SMP_T_STR;
>>>     CID 1539956:  Uninitialized variables  (UNINIT)
>>>     Using uninitialized value "aead_tag.flags".
396                     aead_tag.flags &= ~SMP_F_CONST;
397     
398                     if (!var_set(arg_p[3].data.var.name_hash, arg_p[3].data.var.scope, &aead_tag,
399                                  (arg_p[3].data.var.scope == SCOPE_PROC) ? VF_COND_IFEXISTS : 0)) {
400                             goto err;
401                     }
```

### Output of `haproxy -vv`

```plain
no
```

**Comment by haproxyFred** [comment]
Checked this part with Dragan:
*** CID 1539956:  Uninitialized variables  (UNINIT)
/src/ssl_sample.c: 396 in sample_conv_aes_gcm()

About 
*** CID 1539957:    (CHECKED_RETURN)
/src/sample.c: 4266 in sample_conv_jwt_verify_check()
I am not sure of the logic behind the two calls to vars_check_arg() from sample_conv_jwt_verify_check() because it is supposed to check the arg passe as parameter is a string with a scope (txn,proc...). So the second call always fail with
such a converter:
       jwt_verify(txn.jwt_alg,"hmac key hs256")
as used by reg-tests/jwt/jws_verify.vtc regression test.

Will check this with Rémi soon.

**Comment by rlebreton** [comment]
IIRC, concerning the JWT, the vars_check_args call might fail (as in the example mentioned by @haproxyFred), but it is not considered as an error). The calls are just there to ensure that if a variable was actually used in one of those two parameters, then the 'args' contents are replaced by a VAR type instead of just its name.
We can flag them as intentional.

**Comment by chipitsine** [comment]
shall I mark both 1539956 and 1539957 as intentional?

**Comment by rlebreton** [comment]
1539957 can be marked as intentional, I'll let @haproxyFred answer about 1539956. Thanks!

**Comment by haproxyFred** [comment]
1539956 has already been fixed by Dragan: see  0091692d97407eb96d6cfbb888a254f904182ea9

**Comment by chipitsine** [comment]
thanks!
